src/main/java/org/ccci/gto/persistence/aopalliance/DeadLockRetryMethodInterceptorFactory.java

```diff
@@ -0,0 +1,22 @@
+package org.ccci.gto.persistence.aopalliance;
+
+import static org.ccci.gto.persistence.DeadLockRetry.DEFAULT_ATTEMPTS;
+
+import org.aopalliance.intercept.MethodInterceptor;
+import org.ccci.gto.persistence.DeadLockRetryAspectSupport;
+
+import javax.annotation.Nonnull;
+import java.util.HashMap;
+import java.util.Map;
+
+public class DeadLockRetryMethodInterceptorFactory extends DeadLockRetryAspectSupport {
+    private final Map<String, MethodInterceptor> advice = new HashMap<>();
+
+    public MethodInterceptor getAdvice(@Nonnull final String unitName) {
+        return advice.computeIfAbsent(unitName, this::createAdvice);
+    }
+
+    private MethodInterceptor createAdvice(@Nonnull final String unitName) {
+        return invocation -> wrapJoinPoint(invocation::proceed, unitName, DEFAULT_ATTEMPTS);
+    }
+}
```

Comment:
I'm not following why this class is separate from `DeadLockRetryAdvisor`

Reply:
I separated it out because it didn't feel quite right including it within the Advisor itself based on how an Advisor is composed.

the 3 components here are:
- Advice - logic to run before/after/around a method
- Pointcut - expression that defines which methods to wrap with advice
- Advisor - combines Advice with a Pointcut

The functions in the InterceptorFactory deal exclusively with creating Advice and don't deal at all with pointcuts or advisors. The included Advisor bundles a custom annotation Advice with an annotation specific pointcut and doesn't use anything from the InterceptorFactory, so technically it could just directly extend `DeadLockRetryAspectSupport`. But, in scenarios where neither class is a complete solution (e.g. The Key), I didn't want to have to configure two beans identically.

I could flatten the 2 classes to a single class, but that would always force the annotation based advisor to be active (which incidentally is how the AspectJ implementation currently works)

Reply:
Thanks; I understand the problem now.

I think i'd lean towards forcing the annotation based advisor to be active rather than deepening the class hierarchy, but not strongly.